File: python/hashsetOrHashmap/exercise_1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        pointer = 0
        while pointer<(len(nums)):
            if target-nums[pointer] in hashmap:
                return [hashmap[target-nums[pointer]],pointer]
            if nums[pointer] not in hashmap:
                hashmap[nums[pointer]] = pointer
            pointer = pointer+1
        
        return 0

File: java/hashsetOrHashmap/exercise_1.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numberPosition = new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++){
            if (numberPosition.containsKey(target-nums[i])){
                return new int[] {i,numberPosition.get(target-nums[i])};
            }
            numberPosition.put(nums[i],i);
        }
        return null;   
    }
}